packages/ui/src/store/app/index.ts

```diff
@@ -0,0 +1,60 @@
+import {Context, emptyContext, FilePath} from "../../models";
+
+export type Action =
+  | {
+      type: "app/SET_SIDEBAR_OPEN";
+      value: boolean;
+    }
+  | {
+      type: "app/SET_SELECTED_PATHS";
+      paths: Set<FilePath>;
+    };
+
+export interface State {
+  isSidebarOpen: boolean;
+  selectedPaths: Set<FilePath>;
```

Comment:
It would be nice to avoid using things like `Set` and `Map` - serializable state is much nicer. Use `{[file: FilePath]: boolean}` instead.

Reply:
In #108 I use an array instead of a `Set` because it does need to be serialized (to localstorage).  An object makes sense though. I'll change change it in #108

Reply:
I actually think array makes more sense than an object. If its an object it might be assumed that all of the filepaths are keys and only the selected filepaths are set to true. Instead, only the selected filepaths are even in the object.

What about storing it as an array and converting to Set when we use it?